test/unit/commands.test.ts

```diff
@@ -0,0 +1,152 @@
+  'hex'
+);
+
+const msgBodyNKeyWithInvalidUtf8 = Buffer.from(
+  '0000000000cc020000026e0005000000f09f98ff000477726974654572726f727300a60200000330009e02000010696e646578000000000010636f646500f82a0000036b65795061747465726e000f0000001074657874000100000000036b657956616c756500610100000274657874005201000064e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e298830000026572726d736700f2000000453131303030206475706c6963617465206b6579206572726f7220636f6c6c656374696f6e3a20626967646174612e7465737420696e6465783a20746578745f3120647570206b65793a207b20746578743a202264e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883e29883efbfbd2e2e2e22207d000000106f6b000100000000',
+  'hex'
+);
+
+const invalidUtf8InOtherKeyJSON = {
+  n: '��',
+  writeErrors: [
+    {
+      index: 0,
+      code: 11000,
+      keyPattern: {
+        text: 1
+      },
+      keyValue: {
+        text: 'd☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃'
+      },
+      errmsg:
+        'E11000 duplicate key error collection: bigdata.test index: text_1 dup key: { text: "d☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃☃�..." }'
+    }
+  ],
+  ok: 1
+};
+
+describe('BinMsg', function () {
+  it('should throw error if trying to toggle validation settings for keys other than writeErrors', function () {
+    const binMsgInvalidUtf8ErrorMsg = new BinMsg(
+      Buffer.alloc(0),
+      msgHeader,
+      msgBodyInvalidUtf8WriteErrors
+    );
+    expect(() => binMsgInvalidUtf8ErrorMsg.parse({ validation: { utf8: { n: false } } })).to.throw(
+      Error,
+      'Can only toggle validation settings for writeErrors key'
+    );
+  });
+
+  it('should not throw invalid utf8 error when validation disabled for writeErrors', function () {
```

Comment:
I generally prefer the approach to writing tests thinking of tests as documentation, and the test output should read in a manner that is clear to someone just reading it without looking at the code itself. With this in mind it's always good to put one assertion per `it` block when possible, wrapped in a `context`. This produces some really easy to read output for each exact assertion... So an example for this would to change it to:

```javascript
context('when validation is disabled for writeErrors', function () {
  const binMsgInvalidUtf8ErrorMsg = new BinMsg(
    Buffer.alloc(0),
    msgHeader,
    msgBodyInvalidUtf8WriteErrors
  );
  const options = { validation: { utf8: { writeErrors: false } } };

  it('does not raise an error', function () {
    expect(() =>
      binMsgInvalidUtf8ErrorMsg.parse(options)
    ).to.not.throw();
  });

  it('contains the invalid utf8 in the writeError', function () {
    expect(
      BSON.deserialize(badutf8inputtodeserialize, options)
    ).to.deep.equals(invalidUtf8InWriteErrorsJSON);
  });
});
```

Reply:
What do you think about changing the tests with multiple assertions to follow this format?